### PR TITLE
Get source from requests

### PR DESCRIPTION
**Pull request by rgraber**
## Checklist

1. [x] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation N/A
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE) N/A
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes N/A

## Description

Enable parsing the source (browser and OS) from a request.

## Notes

Adds a new method for parsing out the browser and OS from the HTTP_USER_AGENT header. Relies on a new ua_parser library. Also adds ddt to dev requirements for easier unit tests.

### CODE REVIEW

File: kobo/apps/openrosa/libs/utils/viewer_tools.py [hunk 1]
```diff
@@ -104,6 +105,19 @@ def get_client_ip(request):
     return ip
 
 
+def get_human_readable_client_user_agent(request):
+    """
+    Parse the user-agent into a human-readable <Browser> (<OS>) string
+    """
+    user_agent = request.META.get('HTTP_USER_AGENT', None)
+    if not user_agent or user_agent == '':
+        return "No information available"
```
Comment (noliveleger) [inline kobo/apps/openrosa/libs/utils/viewer_tools.py:114]: `"` vs `'` for strings ;-)

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: kobo/apps/audit_log/models.py [hunk 2]
```diff
@@ -14,6 +14,7 @@ class AuditAction(models.TextChoices):
     PUT_BACK = 'put-back', 'PUT BACK'
     REMOVE = 'remove', 'REMOVE'
     UPDATE = 'update', 'UPDATE'
+    AUTH = 'auth', 'AUTH'
```
Comment (noliveleger) [inline kobo/apps/audit_log/models.py:17]: I think this should be added this pr#5035?

File: kobo/apps/openrosa/apps/viewer/tests/test_viewer_tools.py [hunk 3]
```diff
@@ -19,3 +20,32 @@ def test_get_client_ip(self):
...
+        ("", "No information available"),
+        # missing
+        (None, "No information available")
+    )
+    @unpack
+    def test_client_user_agent(self, ua_string, expected_result):
+        factory = RequestFactory()
+        header = {"HTTP_USER_AGENT": ua_string}
+        request = factory.get("/", **header)
+        result = get_human_readable_client_user_agent(request)
```
Comment (noliveleger) [inline kobo/apps/openrosa/apps/viewer/tests/test_viewer_tools.py:50]: Please use single quotes for strings.

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: kobo/apps/audit_log/models.py (hunk 2 above)
Comment (rgraber) [reply kobo/apps/audit_log/models.py:17]: Good catch. Result of branching shenanigans

**Review by noliveleger — APPROVED** [review]

**Pull request merged**